### PR TITLE
Raise an ValueError when Axes.pie accepts negative values #15923

### PR DESCRIPTION
**Pull request by Inception95**
## PR Summary

Issue #15923
Axes.pie accepts negative values which outputs the wrong pie chart. This issue is caused by not checking the positive and negative of all input wedge sizes of "x" in Axes.pie.

Now it will raise a ValueError, which indicates wedge sizes should be non-negative when feeding negative values into Axes.pie. It is done by adding an if condition right before calculating the portion of all sizes in the "x" `array.

## PR Checklist

- [x] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/axes/_axes.py [hunk 1]
```diff
@@ -2921,6 +2921,10 @@ def pie(self, x, explode=None, labels=None, colors=None,
                 "and will be removed %(removal)s; pass a 1D array instead.")
             x = np.atleast_1d(x.squeeze())
 
+        for part in x:
```
Comment (jklymak) [inline lib/matplotlib/axes/_axes.py]: I think you can use `np.any`?

**Comment by jklymak** [comment]
Thanks for the PR.  This should probably get a test as well...

File: lib/matplotlib/axes/_axes.py (hunk 1 above)
Comment (Inception95) [reply lib/matplotlib/axes/_axes.py]: Thanks, np.any(x<0) is neater!

**Comment by Inception95** [comment]
@jklymak Sure, I will add the test.

**Comment by anntzer** [comment]
Thanks for you contribution, hoping to see you around again :)

**Comment by Inception95** [comment]
@anntzer Happy to help! Thanks for this great opportunity.

**Pull request merged**